app/models/partnership_csv_upload.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class PartnershipCsvUpload < ApplicationRecord
+  has_one_attached :csv
+  # does this need a delivery partner reference ?
+  belongs_to :lead_provider, optional: true
```

Comment:
Why is this optional?

Reply:
The model only has a single attribute, the attachment. In order for it to receive an attachment, it has to exist, before the lead provider is referenced. It's a bit odd, I agree

Reply:
Perhaps we could validate against the lead provider and delivery partner attributes, I'll have a look